File: app/src/main/java/mikazuki/android/app/feelingmatch/model/AutoIncrement.java
package mikazuki.android.app.feelingmatch.model;

import io.realm.Realm;
import io.realm.RealmObject;

/**
 * @author haijimakazuki
 */
public class AutoIncrement {
    public static long newId(Realm realm, Class<? extends RealmObject> clazz) {
        return newIdWithIdName(realm, clazz, "id");
    }

    public static long newIdWithIdName(Realm realm, Class<? extends RealmObject> clazz, String idName) {
        Number max = realm.where(clazz).max(idName);
        max = max != null ? max : 0;
        return max.longValue() + 1;
    }
}

File: app/src/main/java/mikazuki/android/app/feelingmatch/model/AutoIncrement.kt
package mikazuki.android.app.feelingmatch.model

import io.realm.Realm
import io.realm.RealmObject

/**
 * @author haijimakazuki
 */
object AutoIncrement {
    fun newId(realm: Realm, clazz: Class<out RealmObject>): Long {
        return newIdWithIdName(realm, clazz, "id")
    }

    fun newIdWithIdName(realm: Realm, clazz: Class<out RealmObject>, idName: String): Long {
        var max: Number? = realm.where(clazz).max(idName)
        max = if (max != null) max else -1
        return max.toLong() + 1
    }
}
